master/pkg/agent/exit.go

```diff
@@ -82,3 +82,25 @@ const (
 	// AgentError denotes that the agent failed to launch the container.
 	AgentError = FailureType("agent failed to launch the container")
 )
+
+// IsRestartableSystemError checks if the error is caused by the system and
```

Comment:
all this should eventually not really be here since k8s uses these error types too (fine for now, just a note)